File: src/OneBitAndTwoBitCharacters.java
public class OneBitAndTwoBitCharacters {
    public boolean isOneBitCharacter(int[] bits) {
        for (int index = 0 ; index < bits.length ; index++) {
            if (bits[index] == 1) {
                index++;
            } else if (bits[index] == 0 && index == bits.length - 1) {
                return true;
            }
        }
        return false;
    }
}

File: python/one_bit_and_two_bit_characters.py
from typing import List


class Solution:
    def isOneBitCharacter(self, bits: List[int]) -> bool:
        count = 0
        index = len(bits) - 2
        while index >= 0 and bits[index] != 0:
            if bits[index] == 1: count += 1
            index -= 1
        return count % 2 == 0
